pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet
+type SubnetSpec struct {
+	// Size of subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPV4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
+	// Access mode of subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	// +kubebuilder:validation:Enum=private;public;isolated
+	AccessMode string `json:"accessMode,omitempty"`
+	// Subnet CIDR.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=2
+	IPAddresses []string `json:"ipAddresses,omitempty"`
+}
+
+// SubnetCondition defines condition of Subnet.
+type SubnetCondition struct {
+	// Type defines condition type.
+	Type SubnetStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// SubnetStatus defines the observed state of Subnet.
+type SubnetStatus struct {
+	// Logical switch ID.
+	LsID       string            `json:"lsID"`
```

Comment:
This should be NSX-T Subnet path? If so, name it NSXTPath?

Reply:
We use the LS instead of Subnet due to the restore requirements for PodVM on WCP.
But we can change to use path instead of ID.

Reply:
I got the point. The LS ID is needed in the NSX Subnet properties. In the CRD, we can change it to Subnet path. Could we change it to nsxSubnetPath? in VPC status, we have nsxVPCPath.

Reply:
Thanks Xiaopei for the explanation, and change it to nsxtSubnetPath.